user/homework.go

```diff
@@ -57,6 +59,31 @@ func (hw *Homework) Save() error {
 	return nil
 }
 
+// SaveAfterFetch filters out problems for tracks the user has yet to start
+// before saving them all to disk.
+func (hw *Homework) SaveAfterFetch(numArgs int) error {
+	if numArgs == 0 {
```

Comment:
My first thought when seeing this was _can we minimize indentation here_?

How about introducing a `hw.RejectItems()` function or something like that? Maybe it could take the arguments, and exit immediately if there's nothing to reject:

```
func (hw *Homework) RejectSomething(...) {
    if nothingToDoHere {
        return
    }
    // do the work
}

func (hw *Homework) SaveAfterFetch(...) error {
     hw.RejectStuff(...)
     if ln(hw.Items) == 0 {
        return err // that you made up
     }
     return hw.Save()
}
```